Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/json/json.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Marshal delegates to json.Marshal
+// It is only here so this package can be a drop-in for common encoding/json uses
+func Marshal(v interface{}) ([]byte, error) {
+	return json.Marshal(v)
+}
+
+// Unmarshal unmarshals the given data
+// If v is a *map[string]interface{}, numbers are converted to int64 or float64
+func Unmarshal(data []byte, v interface{}) error {
```

Comment:
That's what I mean - I don't want someone switching to this accidentally

On Nov 7, 2015, at 9:56 AM, Jordan Liggitt <notifications@github.com> wrote:

In Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/json/json.go
<https://github.com/openshift/origin/pull/5774#discussion_r44213680>:

> +package json
> +
> +import (
> +	"bytes"
> +	"encoding/json"
> +)
> +
> +// Marshal delegates to json.Marshal
> +// It is only here so this package can be a drop-in for common encoding/json uses
> +func Marshal(v interface{}) ([]byte, error) {
> +	return json.Marshal(v)
> +}
> +
> +// Unmarshal unmarshals the given data
> +// If v is a *map[string]interface{}, numbers are converted to int64 or float64
> +func Unmarshal(data []byte, v interface{}) error {

when you do structured decoding, json looks at the types of the destination
fields, so you don't have the float issue

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5774/files#r44213680>.